File: app/src/main/java/com/bogdanorzea/popularmovies/model/object/Review.java
package com.bogdanorzea.popularmovies.model.object;

import android.os.Parcel;
import android.os.Parcelable;

import com.squareup.moshi.Json;

public class Review implements Parcelable {

    @Json(name = "id")
    public String id;
    @Json(name = "author")
    public String author;
    @Json(name = "content")
    public String content;
    @Json(name = "url")
    public String url;

    protected Review(Parcel in) {
        id = in.readString();
        author = in.readString();
        content = in.readString();
        url = in.readString();
    }

    public static final Creator<Review> CREATOR = new Creator<Review>() {
        @Override
        public Review createFromParcel(Parcel in) {
            return new Review(in);
        }

        @Override
        public Review[] newArray(int size) {
            return new Review[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(id);
        parcel.writeString(author);
        parcel.writeString(content);
        parcel.writeString(url);
    }
}

File: app/src/main/java/com/bogdanorzea/popularmovies/model/object/Review.kt
package com.bogdanorzea.popularmovies.model.`object`

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Review(@Json(name = "id") var id: String,
                  @Json(name = "author") var author: String,
                  @Json(name = "content") var content: String,
                  @Json(name = "url") var url: String) : Parcelable
